src/components/CmsZone.vue

```diff
@@ -275,40 +275,48 @@ export default class CmsZone extends Vue {
     cmsClient.trackZone({ content, zoneId: this.zoneId });
   }
 
-  private trackScrollable(contents: Content[], scrollable: Element): void {
-    if (this.zoneType === 'carousel') {
-      let timeout: number;
-      const listener = (): void => {
-        window.clearTimeout(timeout);
-        timeout = window.setTimeout((): void => {
-          const contentElm = this.$el.querySelector('.slick-current');
-          if (contentElm && this.isContentVisible(contentElm, scrollable, percentVisible)) {
-            this.trackIndex(0);
-            scrollable.removeEventListener('scroll', listener);
-          }
-        }, durationVisibleToBeTrackedMs);
+  private addScrollListener(callback: () => void | Promise<void>, delay: number = 0): () => void {
+    if (delay) {
+      const rawCallback = callback;
+      let timer: number;
+      callback = (): void => {
+        window.clearTimeout(timer);
+        timer = window.setTimeout(rawCallback, delay);
       };
-      scrollable.addEventListener('scroll', listener);
-      this.scrollableListeners.push(listener);
-      Vue.nextTick(listener);
+    }
+
+    const listener = debounce(callback, 100, { leading: true, trailing: true });
+    const unbind = () => document.removeEventListener('scroll', listener, true);
+    document.addEventListener('scroll', listener, true);
```

Comment:
why do we need the last `true` here? i don't think we need to intercept the scroll event before it gets to nested DOM elements do we?

Reply:
Also, I think it might be safer to listen to the closest scrollable container rather than the document directly. I think the document `scroll` event only fires if the body is scrolled but most of the time especially in the crazy vuetify world, it's some other div that's actually scrolling. So maybe something like https://gist.github.com/wojtekmaj/fe811af47fad12a7265b6f7df1017c83?

Reply:
The last true makes it so that this listener is run on any scroll event on any element, without the true, it only listens to scrolling on the body. This feels safer than only listening on the nearest scrollable since it would handle nested scrolling as well. Will checkout the gist and think about it some more.